bin/stress-test/src/store_tests.rs

```diff
@@ -0,0 +1,213 @@
+        .buffer_unordered(concurrency)
+        .map(|res| res.unwrap())
+        .collect::<Vec<_>>()
+        .await;
+
+    print_summary(timers_accumulator);
+}
+
+/// Sends a single `sync_notes` request to the store and returns the elapsed time.
+pub async fn sync_notes(
+    api_client: &mut ApiClient<InterceptedService<Channel, OtelInterceptor>>,
+    account_ids: Vec<AccountId>,
+) -> Duration {
+    let note_tags = account_ids
+        .iter()
+        .map(|id| u32::from(NoteTag::from_account_id(*id, NoteExecutionMode::Local).unwrap()))
+        .collect::<Vec<_>>();
+    let sync_request = SyncNoteRequest { block_num: 0, note_tags };
+
+    let start = Instant::now();
+    api_client.sync_notes(sync_request).await.unwrap();
+    start.elapsed()
+}
+
+// CHECK NULLIFIERS BY PREFIX
+// ================================================================================================
+
+/// Sends multiple `check_nullifiers_by_prefix` requests to the store and prints the performance.
+pub async fn bench_check_nullifiers_by_prefix(
+    data_directory: PathBuf,
+    iterations: usize,
+    concurrency: usize,
+) {
+    let mut store_client = start_store(data_directory.clone()).await;
+
+    let accounts_file = data_directory.join(ACCOUNTS_FILENAME);
+    let accounts = fs::read_to_string(accounts_file).await.unwrap();
+    let account_ids: Vec<AccountId> =
+        accounts.lines().map(|a| AccountId::from_hex(a).unwrap()).collect();
+
+    // get all nullifier prefixes from the store
+    let mut nullifier_prefixes: Vec<u32> = vec![];
+    let mut current_block_num = 0;
+    loop {
+        // get the accounts notes
+        let (_, response) =
+            sync_state(&mut store_client, account_ids.clone(), current_block_num).await;
+        let note_ids = response.notes.iter().map(|n| n.note_id.unwrap()).collect::<Vec<Digest>>();
+
+        // get the notes nullifiers.
+        let notes = store_client
+            .get_notes_by_id(GetNotesByIdRequest { note_ids })
+            .await
+            .unwrap()
+            .into_inner()
+            .notes;
+
+        nullifier_prefixes.extend(
+            notes
+                .iter()
+                .filter_map(|n| {
+                    // private notes are filtered out because `n.details` is None
+                    Note::read_from_bytes(n.details.as_ref()?)
+                        .ok()
+                        .map(|n| u32::from(n.nullifier().prefix()))
+                })
+                .collect::<Vec<u32>>(),
+        );
+
+        current_block_num = response.block_header.unwrap().block_num;
+        if response.chain_tip == current_block_num {
+            break;
+        }
+    }
+    let mut nullifiers = nullifier_prefixes.into_iter().cycle();
+
+    // each request will have 10 prefixes and block number 0
+    let request = |_| {
+        let mut client = store_client.clone();
+
+        let nullifiers_batch: Vec<u32> = nullifiers.by_ref().take(10).collect();
```

Comment:
The number of sent nullifier prefixes is something that we probably want to modify easily so we may want to make it a parameter